plugins/terminal.koplugin/main.lua

```diff
@@ -13,15 +15,51 @@ local WidgetContainer = require("ui/widget/container/widgetcontainer")
 local logger = require("logger")
 local util = require("ffi/util")
 local _ = require("gettext")
+local BaseUtil = require("ffi/util")
 local N_ = _.ngettext
 local Screen = require("device").screen
 local T = util.template
 
+local function getPlaceholder(label)
+    return string.lower(string.sub(label, 1, 1))
```

Comment:
Best to get rid of that function (which get the first char of "KOReader dir" and lowercase it to have `%k` working) and just add the explicite char in your placeholders table:
```lua
    placeholders = {
        {
            -- is it wise to translate these labels? It would have to have consequences for the placeholder strings:
            "directory of current ebook",
+           "d"
            function(command)
                return command:gsub("%%d", BaseUtil.dirname(G_reader_settings:readSetting("lastfile")))
            end
        },
```